tekton/utils.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tekton
+
+import (
+	libhandler "github.com/operator-framework/operator-lib/handler"
+	"github.com/redhat-appstudio/internal-services/api/v1alpha1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+// GetResultsFromPipelineRun returns a map with all the results emitted by the given PipelineRun. Only string results
+// are supported. Other type of results will be silently omitted.
+func GetResultsFromPipelineRun(pipelineRun *tektonv1beta1.PipelineRun) map[string]string {
+	results := map[string]string{}
+
+	for _, pipelineResult := range pipelineRun.Status.PipelineResults {
+		if pipelineResult.Value.Type == tektonv1beta1.ParamTypeString {
+			results[pipelineResult.Name] = pipelineResult.Value.StringVal
+		}
+	}
+
+	return results
+}
+
+// isInternalRequestsPipelineRun returns a boolean indicating whether the object passed is an internal request
+// PipelineRun or not.
+func isInternalRequestsPipelineRun(object client.Object) bool {
+	_, ok := object.(*tektonv1beta1.PipelineRun)
+	if !ok {
+		return false
+	}
+
+	internalRequestKind := reflect.TypeOf(v1alpha1.InternalRequest{}).Name()
+
+	if ownerType, ok := object.GetAnnotations()[libhandler.TypeAnnotation]; ok {
+		return strings.Contains(ownerType, internalRequestKind)
+	}
+
+	return false
+}
+
+// hasPipelineSucceeded returns a boolean indicating whether the PipelineRun succeeded or not.
```

Comment:
Can you improve this comment? Two objects are passed in and the comment sounds like just one is

Reply:
I still think this comment relies on decent knowledge of predicates. Maybe `returns a boolean that is true if objectOld has not yet succeeded and objectNew has` or something along those lines. There is no "the" pipelinerun. You could pass two totally different pipelineruns to this function... understanding it relies on knowing that predicate update events give you an old and new